prov/tcp/src/tcpx_rma.c

```diff
@@ -107,7 +107,71 @@ static ssize_t tcpx_rma_readv(struct fid_ep *ep, const struct iovec *iov, void *
 static ssize_t tcpx_rma_writemsg(struct fid_ep *ep, const struct fi_msg_rma *msg,
 		uint64_t flags)
 {
-	return -FI_ENODATA;
+	struct tcpx_ep *tcpx_ep;
+	struct tcpx_cq *tcpx_cq;
+	struct tcpx_pe_entry *send_entry;
+	uint64_t data_len;
+
+	tcpx_ep = container_of(ep, struct tcpx_ep, util_ep.ep_fid);
+	tcpx_cq = container_of(tcpx_ep->util_ep.tx_cq, struct tcpx_cq,
+			       util_cq);
+
+	send_entry = tcpx_pe_entry_alloc(tcpx_cq);
+	if (!send_entry)
+		return -FI_EAGAIN;
+
+	assert(msg->iov_count <= TCPX_IOV_LIMIT);
```

Comment:
This assert is <=, but the rma_iov_count assert below uses <.  < seems more appropriate.